### ISSUE TITLE
oce.edit() doc error and request re data/metadata

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1361]
* [ ] The doc says
> 3. Applied to an adv object (i.e. data from an acoustic velocimeter), oceEdit treats items named heading, pitch, roll appropriately, depending on the type of adv instrument used. (This is necessary because different manufacturers produce different forms of these items, i.e. Nortek reports them on a time base that is different from the velocity reporting, while Sontek reports them on the same time base.)
which is apparently a remnant from `adv` docs.

* [ ] It would be nice to have a syntax to force the change to `data` or `metadata`, in case the default chosen by `oce.edit` is not desired. This could be a flag `destination` with default `"auto"` meaning to do as is done now, or `"data"` or `"metadata"` to work therein.

I'm marking this as low-priority because this has never come up, but I thought of it whilst editing a section in OAR.

**Comment by dankelley** [comment]
Fixed in "develop" branch commit 75fb082e5d0196aeecc7f2220f282c352071e9d8 and I've added some tests to `tests/testthat/test_misc.R`, viz. the below, that ensure that the rules are followed. The rule is basically that things work like before, but that e.g. `oceEdit(ctd, "data@something", newSomething)` will specifically alter the named item in the named slot. Also, as a new feature, if there is a `@` in `item`, then we can **add** something that didn't exist before. The point of this `@` is that it prevents `oceEdit` from having to guess whether to put some new thing in `metadata` or in `data`.
```R
test_that("oceEdit", {
          data(ctd)
          ## metadata
          a <- oceEdit(ctd, "waterDepth", 100)
          expect_equal(a@metadata$waterDepth, 100)
          b <- oceEdit(a, "waterDepth", 200)
          expect_equal(b@metadata$waterDepth, 200)
          c <- oceEdit(b, "metadata@waterDepth", 300)
          expect_equal(c@metadata$waterDepth, 300)
          ## data
          scan <- ctd[["scan"]]
          ctd2 <- oceEdit(ctd, "scan", scan+1)
          expect_true("scan" %in% names(ctd2[["data"]]))
          expect_false("scan" %in% names(ctd2[["metadata"]]))
          expect_equal(ctd[["scan"]]+1, ctd2[["scan"]])
          ## check "case 1A" of docs (cannot add non-extant item)
          expect_error(oceEdit(ctd, "noSuchThing", 1), "nothing named 'noSuchThing'")
          ## check "case 2" of docs (can add non-extant item, if slot provided)
          ctd2 <- oceEdit(ctd, "metadata@newMetadata", 1)
          expect_true("newMetadata" %in% names(ctd2[["metadata"]]))
          ctd2 <- oceEdit(ctd, "data@newData", 1)
          expect_true("newData" %in% names(ctd2[["data"]]))
})
```

**Comment by dankelley** [comment]
PS. forgot to say -- I simplified the code. There is still some special-casing for ADV and ADP types, and I am not at all sure that is doing useful things. I think it's a remnant from long-ago days, when we didn't even have slots (i.e. before the transition to S4 objects). So far as I can guess, the old code was trying to let a person name an item `heading` and then `oceEdit` would figure out if it was a "fast" or a "slow" heading ... but the code doesn't seem to be doing that, now. If @richardsc is worried about this (since he is familiar with the ADV and ADP nuances), perhaps he'll reopen this and add some comments that include names of SLEIWEX files that I can use to check on things.
